File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    sum = 0
    for i in range(1, len(s) +1):
        if(s[-i] == ')'):
            sum += 1
        elif(s[-i] == '('):
            sum -= 1
        if(sum <0):
            return False
    
    if(sum == 0):
        return True
    else:
        return False
    
    

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include <bits/stdc++.h>

using namespace std;

bool solution(string s)
{
    // stack 만들기 //!!항상 뭐 정의할 때 타입(int, string, ...)도 같이 써줘야 돼
    stack<char> st;
    
    // ( 이면 넣고  
    // ) 이면 빼기
    for (char a: s){
        if (a == '('){
            st.push(a);
        }
        else{
            // 없는데 빼려고 하면 false return
            if(st.size()>0){
                st.pop();
            }
            else{
                return false;
            }
        }
    }
    // s 끝까지 보고 stack에도 남은 거 없으면 true return
    if(st.empty()){
        return true;
    } else{
        return false;
    }
}